src/odemis/gui/cont/acquisition/cryo_acq.py

```diff
@@ -690,3 +748,54 @@ def _on_features_change(self, features):
             self._panel.btn_acquire_features.SetToolTip(DISABLED_TOOLTIP)
             self._panel.chk_use_autofocus_acquire_features.Disable()
             self._panel.chk_use_autofocus_acquire_features.SetToolTip(DISABLED_TOOLTIP)
+
+        self._update_features_chklist(features)
+        self._update_acquisition_time()
+
+    @call_in_wx_main
+    def _update_selected_feature(self, evt: wx.Event):
+        # get the index of selected item, and update the current feature
+        index = self._panel.acquire_features_chk_list.GetSelection()
+        f = self._tab_data.main.features.value[index]
+        self._tab_data.main.currentFeature.value = f
+
+    def _update_checked_features(self, evt: wx.Event):
+        """Prevent the user from checking a disabled feature."""
+        index = evt.GetInt()
+        disabled_features_indexes = [
+            i
+            for i, f in enumerate(self._tab_data.main.features.value)
+            if f.status.value in [FEATURE_DEACTIVE]
+        ]
+
+        # Prevent the change
+        if index in disabled_features_indexes:
+            self._panel.acquire_features_chk_list.Check(index, False)
+            f = self._tab_data.main.features.value[index]
+            disabled_txt = f"{f.name.value} is Discarded and cannot be acquired."
+            wx.MessageBox(disabled_txt, "Info", wx.OK | wx.ICON_INFORMATION)
+
+        # Update the acquisition time
+        self._update_acquisition_time()
+
+    def _update_feature_status(self, feature: CryoFeature):
+        self._update_features_chklist(self._tab_data.main.features.value)
+
+    @call_in_wx_main
+    def _update_features_chklist(self, features: List[CryoFeature]):
+        """
+        Sync the features with the chklistbox
+        """
+        # clear the list
+        self._panel.acquire_features_chk_list.Clear()
+
+        # add the features to the list, and check the enabled ones
+        for i, f in enumerate(features):
+            txt = f"{f.name.value} ({f.status.value})"
+            check = False if f.status.value in DISABLED_FEATURE_STATUS else True
```

Comment:
Can't you not use a list for DISABLED_FEATURE_STATUS, that would simplify things:
```suggestion
            check = f.status.value != DISABLED_FEATURE_STATUS
```